### PR TITLE
Adding void instance

### LINKED ISSUE #46: Generic instance for Void
Would it make sense to have a generic instance for `Void`, similar to the instance for `Unit`?

### PR DESCRIPTION
**Pull request by ethul**
Resolves #46

### CODE REVIEW

**Comment by paf31** [comment]
Sorry, I don't think my brain is working today. I should have suggested just not adding an `SVoid` constructor at all 😄 Now I see what @garyb was getting at. And you could implement `fromSpine` with `absurd`.

**Comment by ethul** [comment]
That works for me. Let me make the change.

**Comment by LiamGoodacre** [comment]
@paf31 How can you implement `fromSpine` using `absurd` without having a `Void` to pass it?

**Comment by ethul** [comment]
I think `toSpine` would use `absurd`

**Comment by LiamGoodacre** [comment]
Ah I see

File: src/Data/Generic.purs [hunk 1]
```diff
@@ -277,6 +282,7 @@ data GenericSignature
   | SigChar
   | SigArray (Unit -> GenericSignature)
   | SigUnit
+  | SigVoid
```
Comment (paf31) [inline src/Data/Generic.purs]: We can do this without a breaking change if we use `SigProd "Data.Void.Void" []` here.

Comment (ethul) [inline src/Data/Generic.purs]: Sounds good!

**Comment by paf31** [comment]
Great, thanks!

**Comment by ethul** [comment]
Welcome, thank you!

**Pull request merged**